core/src/main/javascript/reportviewer/reportviewer.js

```diff
@@ -135,7 +135,9 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
           this._locationPromptAttachHandlerRegistration = window.top.mantle_addHandler('GenericEvent', onAttach.bind(this) );
         }
 
-        window.onbeforeunload = this.dispose.bind(this);
+        // The added callback snippet, that apparently doesn't do anything, will determine the decision to whether the
+        // XMLHttpRequest call is an async one or not (in common-ui's pentaho-ajax.js#L308-L310).
+        window.onbeforeunload = this.dispose.bind(this, function() {});
```

Comment:
Checkout [BACKLOG-34390](https://jira.pentaho.com/browse/BACKLOG-34390?focusedCommentId=399798&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-399798) for a more complete explanation on this change